client.go

```diff
@@ -784,9 +802,17 @@ func (c *Client) establishRegion(originalReg *regioninfo.Info, host string, port
 			}
 		}
 		if c.clientType == AdminClient {
-			host, port, err = c.zkLookup(ctx, zk.Master)
+			if c.master == (zk.ResourceName)("") {
```

Comment:
Then you don't need a test here, right? After you test this in the setter funcs. :smile: